app/controllers/api/v1/questions_controller.rb

```diff
@@ -0,0 +1,136 @@
+class Api::V1::QuestionsController < ApplicationController
+  
+  # Index method returns the list of questions JSON object
+  # GET on /questions
+  def index
+    @questions = Question.order(:id)
+    render json: @questions, status: :ok
+  end
+
+  # Show method returns the question object with id - {:id}
+  # GET on /questions/:id
+  def show
+    begin
+      @question = Question.find(params[:id])
+      render json: @question, status: :ok
+    rescue ActiveRecord::RecordNotFound
+      render json: $ERROR_INFO.to_s, status: :not_found
+    end
+  end
+
+  # Create method returns the JSON object of created question
+  # POST on /questions
+  def create
```

Comment:
Here is a possible refactor:

```ruby
def create
  questionnaire_id = params[:questionnaire_id]
  questionnaire = Questionnaire.find(questionnaire_id)
  question = questionnaire.questions.build(
    txt: params[:txt],
    question_type: params[:question_type],
    break_before: true
  )

  case question.question_type
  when 'Scale'
    question.weight = params[:weight]
    question.max_label = 'Strongly agree'
    question.min_label = 'Strongly disagree'
  when 'Cake', 'Criterion'
    question.weight = params[:weight]
    question.max_label = 'Strongly agree'
    question.min_label = 'Strongly disagree'
    question.size = '50, 3'
  when 'Dropdown'
    question.alternatives = '0|1|2|3|4|5'
  when 'TextArea'
    question.size = '60, 5'
  when 'TextField'
    question.size = '30'
  end

  if question.save
    render json: question, status: :created
  else
    render json: question.errors.full_messages.to_sentence, status: :unprocessable_entity
  end
rescue ActiveRecord::RecordNotFound
  render json: $ERROR_INFO.to_s, status: :not_found and return  
end
```

```ruby
class Question < ApplicationRecord
  before_create :set_seq
  belongs_to :questionnaire

  def set_seq
    self.seq = questionnaire.questions.size + 1
  end
end
```

1. The num_of_existed_questions variable is no longer needed since it can be inferred from the model's questionnaire.questions.size.
2. Instead of creating a new question and setting its attributes one by one, we can use build to create a new instance of Question associated with the questionnaire.
3. We can simplify the case statement by removing unnecessary question.weight and question.size assignments that are already covered by other cases. Also, we don't need to set question.size to nil in the 'Dropdown' case since it is already nil by default.
4. We can simplify the error handling by checking question.save instead of rescuing from ActiveRecord::RecordInvalid. This also allows us to access the error messages using question.errors.full_messages.

Reply:
This change requires additional tests to be done before this could be accomodated in the code. We would need extra time to resolve this issue.